Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/land.py

```diff
@@ -70,10 +71,46 @@ def parser(cls, parser, loggers=None):
             action=arguments.NoAction,
         )
         parser.add_argument(
-            '--defaults', '--no-defaults', action=arguments.NoAction, default=None,
-            help='Do not prompt the user for defaults, always use (or do not use) them',
+            '--safe', '--unsafe',
+            dest='safe', default=None,
+            help='Land change via safe (or unsafe) merge queue, if available.',
+            action=arguments.NoAction,
         )
 
+    @classmethod
+    def merge_queue(cls, args, repository, branch_point, merge_labels=None):
+        log.info('Detected merging automation, using that instead of local git tooling')
+        merge_type = {
+            True: 'safe',
+            False: 'unsafe',
+        }.get(args.safe, sorted(merge_labels.keys())[0])
+        merge_label = merge_labels.get(merge_type)
+        if not merge_label:
+            sys.stderr.write("No {} merge-queue available for this repository\n".format(merge_type))
+            return 1
+
+        def callback(pr):
+            pr_issue = pr._metadata.get('issue')
+            if not pr_issue:
+                sys.stderr.write("Cannot set any labels on '{}' because the service doesn't support labels\n".format(pr))
+                return 1
+
+            labels = pr_issue.labels
+            if PullRequest.BLOCKED_LABEL in labels:
```

Comment:
Are we sure about this ? merge-blocked maybe added by EWS, should we do this only for unsafe merge ?

Reply:
I am fine with other part, r+ with comments.